### ISSUE TITLE
Dynamically generated Quarto blog post titles

### ISSUE THREAD

**Issue by DrewArnold1** [quarto-dev/quarto-web#126]
I am creating a blog that automatically creates a new post everyday by rendering a Quarto document.  The same Quarto document is rendered everyday, but is rendered with data that has been generated over the past 24 hours, making it a fresh post.

I'm having an issue dynamically generating titles and dates.  Ideally, I would use R code to specify titles and dates.  My current attempt is as follows:

```
---
title: "`r paste(lubridate::today(), 'Report')`"
date: "today"
---
```

Interestingly, the title appears as the R code's output in the post itself, but the listing page displays the R code as the title instead of its output.  Is it possible to have the R code's output displayed as the title on the listing page?

On the dynamic date issue, I have seen this [callout](https://github.com/quarto-dev/quarto-web/blob/main/docs/websites/website-blog.qmd#L71) warning not to use dynamic dates in blog posts.  Is there a solution for my use case?  Could I pass a dynamic date into the Quarto document as a param?

Thanks for the help, and great job with Quarto!  It's a phenomenal tool.

**Comment by jjallaire** [comment]
The problem is that things like listings examine titles as text prior to rendering, so making them dynamic is difficult. One solution would be to examine the HTML output rather than the markdown source, but that's kind of an inversion of what exists now so not an easy fix for us.

You might want to orchestrate this by having a "template" qmd file which you create a copy of each day.

**Comment by DrewArnold1** [comment]
@jjallaire That was my thought, too.  I orchestrate this by creating a copy of a template each day.  My daily script looks like this:

```
new_post_dir_name = paste0("posts/", lubridate::today())
new_post_path = paste0(new_post_dir_name, "/index.qmd")

dir.create(new_post_dir_name)

file.copy(
  from = "_daily_post_template.qmd",
  to = new_post_path,
  overwrite = TRUE
)

quarto::quarto_render()
```

Is there a way I could adjust this script, or parameters that I could add, that would allow the post titles and dates to be generated dynamically?

Thanks for your response.

**Comment by jjallaire** [comment]
Yes, you can provide metadata externally using `metadata-file`. So for example in your main post document:

```yaml
---
metadata-file: titleblock.yml
---
```

Then just generate `titleblock.yml` dynamically

**Comment by DrewArnold1** [comment]
Beautiful, that works great.  I've pasted the updated daily script below in case someone stumbles across this post in the future.  Thanks for your help!

```
new_post_dir_name = paste0("posts/", lubridate::today())
new_post_path = paste0(new_post_dir_name, "/index.qmd")

dir.create(new_post_dir_name)

post_title = paste(lubridate::today(), "Report")
post_date = paste(lubridate::today())
attr(post_title, "quoted") = TRUE
attr(post_date, "quoted") = TRUE

yaml::write_yaml(
  list(
    title = post_title, 
    date = post_date
  ),
  file = paste0(new_post_dir_name, "/_dynamic_title_and_date.yml")
)

file.copy(
  from = "_daily_post_template.qmd",
  to = new_post_path,
  overwrite = TRUE
)

quarto::quarto_render()
```

**Comment by DrewArnold1** [comment]
@jjallaire There's actually a side effect to specifying date in a separate yaml file ... Quarto does not pick up that date for the listing page or the RSS feed.  As a result, it uses the build datetime as the date (see [here](https://github.com/quarto-dev/quarto-cli/blob/af5c051e8251d3a0535f2e8dcf62d011259c8fb2/src/project/types/website/listing/website-listing-feed.ts#L399)).

Is there a way around this?

**Comment by jjallaire** [comment]
@dragonstyle We may need to fully resolve any `metadata-file` references in the front matter before we pick the date out for listings

**Comment by dragonstyle** [comment]
Just so! Fixed here: b064bec1efe7af4e3332c74f699686480baead12.  A fresh release is building and it will be available as of Quarto v0.9 build 330. Let us know if this gets you going!

**Comment by DrewArnold1** [comment]
You all move fast, thanks for this!
